File: AdvancedExample/AdvancedExample/ViewController.m
//
//  ViewController.m
//  HeadsAppSampleApp
//
//  Created by HeadsApp on 9/2/15.
//  Copyright (c) 2015 HeadsApp. All rights reserved.
//

#import "ViewController.h"
#import <HeadsAppSDK/HeadsAppSDK.h>
#import "Constants.h"

//Fullscreen Video ADs placement IDs

#define VIDEO_PLACEMENT_ID1 @"VIDEO_PLACEMENT_ID1"

//Fullscreen Vungle Rewarded Video ADs placement IDs
#define VUNGLE_REWARDED_VIDEO_PLACEMENT_ID @"VUNGLE_REWARDED_VIDEO_PLACEMENT_ID"

//Fullscreen AdColony Rewarded Video ADs placement IDs
#define ADCOLONY_REWARDED_VIDEO_PLACEMENT_ID @"ADCOLONY_REWARDED_VIDEO_PLACEMENT_ID"

//Other Fullscreen ADs placement IDs
#define INTERSTITIAL_PLACEMENT_ID1 @"INTERSTITIAL_PLACEMENT_ID1"

//Banner ADs placement IDs
#define BANNER_PLACEMENT_ID1 @"BANNER_PLACEMENT_ID1"

@interface ViewController ()

//banner AD buttons
@property (weak, nonatomic) IBOutlet UIButton *bannerAdButton1;

//video AD buttons
@property (weak, nonatomic) IBOutlet UIButton *videoAdButton1;
@property (weak, nonatomic) IBOutlet UIButton *vungleRewardedVideoAdButton;
@property (weak, nonatomic) IBOutlet UIButton *adColonyRewardedVideoAdButton;

//other fullscreen AD buttons
@property (weak, nonatomic) IBOutlet UIButton *interstitialAdButton1;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    [self.videoAdButton1 setEnabled:NO];
    [self.bannerAdButton1 setEnabled:NO];
    [self.interstitialAdButton1 setEnabled:NO];
    [self.vungleRewardedVideoAdButton setEnabled:NO];
    [self.adColonyRewardedVideoAdButton setEnabled:NO];
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(videoAdAvailabilityChanged:) name:kAdAvailabilityChangedNotification object:nil];
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(fullscreenAdClosedWithProviderInfo:) name:kAdVideoClosedNotification object:nil];
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
    
    [self.videoAdButton1 setEnabled:[[HeadsAppSDK sharedSDK] isAdPlayableForPlacementID:VIDEO_PLACEMENT_ID1]];
    [self.bannerAdButton1 setEnabled:[[HeadsAppSDK sharedSDK] isAdPlayableForPlacementID:BANNER_PLACEMENT_ID1]];
    [self.interstitialAdButton1 setEnabled:[[HeadsAppSDK sharedSDK] isAdPlayableForPlacementID:INTERSTITIAL_PLACEMENT_ID1]];
    [self.vungleRewardedVideoAdButton setEnabled:[[HeadsAppSDK sharedSDK] isAdPlayableForPlacementID:VUNGLE_REWARDED_VIDEO_PLACEMENT_ID]];
    [self.adColonyRewardedVideoAdButton setEnabled:[[HeadsAppSDK sharedSDK] isAdPlayableForPlacementID:ADCOLONY_REWARDED_VIDEO_PLACEMENT_ID]];
    
}

- (void)dealloc
{
    [[NSNotificationCenter defaultCenter] removeObserver:self];
}

- (IBAction)requestBannerAd1:(id)sender {
    [[HeadsAppSDK sharedSDK] playAdForPlacementID:BANNER_PLACEMENT_ID1 fromViewController:self];
}

- (IBAction)requestVideoAd1:(id)sender {
    [[HeadsAppSDK sharedSDK] playAdForPlacementID:VIDEO_PLACEMENT_ID1 fromViewController:self];
}

- (IBAction)requestVungleRewardedVideoAd:(id)sender {
    [[HeadsAppSDK sharedSDK] playAdForPlacementID:VUNGLE_REWARDED_VIDEO_PLACEMENT_ID fromViewController:self];
}

- (IBAction)requestAdColonyRewardedVideoAd:(id)sender {
    [[HeadsAppSDK sharedSDK] playAdForPlacementID:ADCOLONY_REWARDED_VIDEO_PLACEMENT_ID fromViewController:self];
}

- (IBAction)requestInterstitialAd1:(id)sender {
    [[HeadsAppSDK sharedSDK] playAdForPlacementID:INTERSTITIAL_PLACEMENT_ID1 fromViewController:self];
}

#pragma mark - Video Ads buttons

- (void)videoAdAvailabilityChanged:(NSNotification *)notification {
    NSDictionary *info = [notification userInfo];
    if ([info isKindOfClass:[NSDictionary class]]) {
        NSString *placementId = [info objectForKey:kPlacementIDKey];
        BOOL availableNumber = [[info objectForKey:kAdAvailableKey] boolValue];
        
        if ([placementId isEqualToString:VIDEO_PLACEMENT_ID1]) {
            [self.videoAdButton1 setEnabled:availableNumber];
        } else if ([placementId isEqualToString:BANNER_PLACEMENT_ID1]) {
            [self.bannerAdButton1 setEnabled:availableNumber];
        } else if ([placementId isEqualToString:INTERSTITIAL_PLACEMENT_ID1]) {
            [self.interstitialAdButton1 setEnabled:availableNumber];
        } else if ([placementId isEqualToString:VUNGLE_REWARDED_VIDEO_PLACEMENT_ID]) {
            [self.vungleRewardedVideoAdButton setEnabled:availableNumber];
        } else if ([placementId isEqualToString:ADCOLONY_REWARDED_VIDEO_PLACEMENT_ID]) {
            [self.adColonyRewardedVideoAdButton setEnabled:availableNumber];
        }
    }
}

#pragma mark - Ad closed and rewarded video details

- (void)fullscreenAdClosedWithProviderInfo:(NSNotification *)notification {
    NSDictionary *viewInfo = [notification userInfo];
    
    //First check whether the video was fully played
    if (viewInfo != nil && [[viewInfo objectForKey:HeadsAppVideoAdFinishedKey] boolValue] == YES) {
        //If last presented AD is Vungle rewarded video
        if ([[viewInfo objectForKey:HeadsAppPlayAdOptionKeyPlacementID] isEqualToString:VUNGLE_REWARDED_VIDEO_PLACEMENT_ID]) {
            //Display reward to the user
            UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"You've been rewarded" message:@"You've been rewarded for playing the video" preferredStyle:UIAlertControllerStyleAlert];
            UIAlertAction *okAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
                [alert dismissViewControllerAnimated:YES completion:^{
                }];
            }];
            [alert addAction:okAction];
            [self presentViewController:alert animated:YES completion:^{
                
            }];
        }
        //If last presented AD is AdColony rewarded video check the reward details for the HeadsAppPlayAdOptionAdColonyCurrencyName and HeadsAppPlayAdOptionAdColonyCurrencyAmount keys
        else if ([[viewInfo objectForKey:HeadsAppPlayAdOptionKeyPlacementID] isEqualToString:ADCOLONY_REWARDED_VIDEO_PLACEMENT_ID]) {
            NSDictionary *rewardedInfo = [viewInfo objectForKey:HeadsAppAdProviderInfoKey];
            NSLog(@"Additional reward info: %@", rewardedInfo);

            NSInteger coins = [[viewInfo objectForKey:HeadsAppPlayAdOptionAdColonyCurrencyAmount] integerValue];
            NSString *currency = [viewInfo objectForKey:HeadsAppPlayAdOptionAdColonyCurrencyName];
            
            NSString *message = nil;
            if ([currency isKindOfClass:[NSString class]] && [currency length] > 0) {
                message = [NSString stringWithFormat:@"You've been rewarded with %d %@ for playing the video", (int)coins, currency];
            } else {
                message = [NSString stringWithFormat:@"You've been rewarded with %d for playing the video", (int)coins];
            }
            
            UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"You've been rewarded" message:message preferredStyle:UIAlertControllerStyleAlert];
            UIAlertAction *okAction = [UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:^(UIAlertAction * _Nonnull action) {
            }];
            [alert addAction:okAction];
            [self presentViewController:alert animated:YES completion:^{
            }];
        }
    }
}

@end

File: AdvancedSwiftExample/AdvancedSwiftExample/ViewController.swift
//
//  ViewController.swift
//  AdvancedSwiftExample
//
//  Copyright © 2016 HeadsApp. All rights reserved.
//

import UIKit
import HeadsAppSDK

//Fullscreen Video ADs placement IDs 
let VIDEO_PLACEMENT_ID1 = "VIDEO_PLACEMENT_ID1"

//Fullscreen Vungle Rewarded Video ADs placement IDs
let VUNGLE_REWARDED_VIDEO_PLACEMENT_ID = "VUNGLE_REWARDED_VIDEO_PLACEMENT_ID"

//Fullscreen AdColony Rewarded Video ADs placement IDs
let ADCOLONY_REWARDED_VIDEO_PLACEMENT_ID = "ADCOLONY_REWARDED_VIDEO_PLACEMENT_ID"

//Other Fullscreen ADs placement IDs
let INTERSTITIAL_PLACEMENT_ID1 = "INTERSTITIAL_PLACEMENT_ID1"

//Banner ADs placement IDs
let BANNER_PLACEMENT_ID1 = "BANNER_PLACEMENT_ID1"


class ViewController: UIViewController {
    
    //banner AD buttons
    @IBOutlet var bannerAdButton1 : UIButton!
    
    //video AD buttons
    @IBOutlet var videoAdButton1 : UIButton!
    @IBOutlet var vungleRewardedVideoAdButton : UIButton!
    @IBOutlet var adColonyRewardedVideoAdButton : UIButton!
    
    //other fullscreen AD buttons
    @IBOutlet var interstitialAdButton1 : UIButton!

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        videoAdButton1.isEnabled = false;
        bannerAdButton1.isEnabled = false;
        interstitialAdButton1.isEnabled = false;
        vungleRewardedVideoAdButton.isEnabled = false;
        adColonyRewardedVideoAdButton.isEnabled = false;
        NotificationCenter.default.addObserver(self, selector: #selector(ViewController.videoAdAvailabilityChanged(_:)), name: NSNotification.Name(rawValue: Constants.kAdAvailabilityChangedNotification), object: nil)
        NotificationCenter.default.addObserver(self, selector: #selector(ViewController.fullscreenAdClosedWithProviderInfo(_:))    , name: NSNotification.Name(rawValue: Constants.kAdVideoClosedNotification), object: nil)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    deinit {
        NotificationCenter.default.removeObserver(self)
    }
    
    
    @IBAction func requestBannerAd1(_ sender: UIButton) {
        HeadsAppSDK.shared()?.playAd(forPlacementID: BANNER_PLACEMENT_ID1, from: self)
    }
    
    @IBAction func requestVideoAd1(_ sender: UIButton) {
        HeadsAppSDK.shared()?.playAd(forPlacementID: VIDEO_PLACEMENT_ID1, from: self)
    }
    
    @IBAction func requestVungleRewardedVideoAd(_ sender: UIButton) {
        HeadsAppSDK.shared()?.playAd(forPlacementID: VUNGLE_REWARDED_VIDEO_PLACEMENT_ID, from: self)
    }
    
    @IBAction func requestAdColonyRewardedVideoAd(_ sender: UIButton) {
        HeadsAppSDK.shared()?.playAd(forPlacementID: ADCOLONY_REWARDED_VIDEO_PLACEMENT_ID, from: self)
    }
    
    @IBAction func requestInterstitialAd1(_ sender: UIButton) {
        HeadsAppSDK.shared()?.playAd(forPlacementID: INTERSTITIAL_PLACEMENT_ID1, from: self)
    }

    //MARK: - Video Ads buttons
    
    func videoAdAvailabilityChanged(_ notification : Notification) {
        if let info = (notification as NSNotification).userInfo {
            if let placementId = info[Constants.kPlacementIDKey] as? String {
                if let availableNumber = info[Constants.kAdAvailableKey] as? Bool {
                    if (placementId == VIDEO_PLACEMENT_ID1) {
                        self.videoAdButton1.isEnabled = availableNumber;
                    } else if (placementId == BANNER_PLACEMENT_ID1) {
                        self.bannerAdButton1.isEnabled = availableNumber;
                    } else if (placementId == INTERSTITIAL_PLACEMENT_ID1) {
                        self.interstitialAdButton1.isEnabled = availableNumber;
                    } else if (placementId == VUNGLE_REWARDED_VIDEO_PLACEMENT_ID) {
                        self.vungleRewardedVideoAdButton.isEnabled = availableNumber;
                    } else if (placementId == ADCOLONY_REWARDED_VIDEO_PLACEMENT_ID) {
                        self.adColonyRewardedVideoAdButton.isEnabled = availableNumber;
                    }
                }
            }
        }
    }
    
   //MARK: - Ad closed and rewarded video details

    func fullscreenAdClosedWithProviderInfo(_ notification: Notification) {
        if let info = (notification as NSNotification).userInfo {
            let rewardedInfo = info[HeadsAppAdProviderInfoKey]
        
            //First check whether the video was fully played
            if let finished = info[HeadsAppVideoAdFinishedKey] as? Bool {
                if finished == true {
                    if let placementId = info[HeadsAppPlayAdOptionKeyPlacementID] as? String {
                        //If last presented AD is Vungle rewarded video
                        if placementId == VUNGLE_REWARDED_VIDEO_PLACEMENT_ID {
                            //Display reward to the user
                            let alert = UIAlertController(title: "You've been rewarded", message: "You've been rewarded for playing the video", preferredStyle: UIAlertControllerStyle.alert)
                            let okAction = UIAlertAction(title: "OK", style: UIAlertActionStyle.default, handler: { (action: UIAlertAction) in
                                alert.dismiss(animated: true, completion: {
                                });
                            })
                            alert.addAction(okAction);
                            self.present(alert, animated: true, completion: {
                            });
                        }
                        //If last presented AD is AdColony rewarded video check the reward details for the HeadsAppPlayAdOptionAdColonyCurrencyName and HeadsAppPlayAdOptionAdColonyCurrencyAmount keys
                        else if placementId == ADCOLONY_REWARDED_VIDEO_PLACEMENT_ID {
                            if rewardedInfo != nil {
                                print("Additional reward info: \(rewardedInfo)");
                                
                                let coins = info[HeadsAppPlayAdOptionAdColonyCurrencyAmount]
                                if let currency : String = info[HeadsAppPlayAdOptionAdColonyCurrencyName] as? String {
                                    var message = "";
                                    if (coins != nil) {
                                        if currency.characters.count > 0 {
                                            message = "You've been rewarded with \(coins!) \(currency) for playing the video"
                                        } else {
                                            message = "You've been rewarded with \(coins!) for playing the video"
                                        }
                                    }
                                    let alert = UIAlertController(title: "You've been rewarded", message: message, preferredStyle: UIAlertControllerStyle.alert)
                                    let okAction = UIAlertAction(title: "OK", style: UIAlertActionStyle.default, handler: { (action: UIAlertAction) in
                                        alert.dismiss(animated: true, completion: {
                                        });
                                    })
                                    alert.addAction(okAction);
                                    self.present(alert, animated: true, completion: {
                                    })
                                }
                                
                            }
                        }
                    }
                }
            }
        }
    }
}
